streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -152,10 +152,6 @@ Consumer<byte[], byte[]> mainConsumer() {
         return mainConsumer;
     }
 
-    StreamsProducer streamsProducerForTask(final TaskId taskId) {
```

Comment:
This is not for this PR itself, but: we can do some further simplifications around TaskManager, e.g. the logic around `shouldCommitAdditionalTasks` was mainly for EOS-v1 since only in that case we may not want to commit for all tasks, for ALOS and EOS-v2, whenever we want to commit, we should always commit all tasks anyways. With V1 removed we should consider reducing the complexity for such cases as well, i.e. when we commit, we are always trying to commit all active tasks except for the error handling case.

Reply:
Sounds good. I will create a separate PR for this.